Source/Classes/PINRemoteImageManager.m

```diff
@@ -36,10 +36,10 @@
 #endif
 
 
-#define PINRemoteImageManagerDefaultTimeout     30.0
-#define PINRemoteImageMaxRetries                3
-#define PINRemoteImageRetryDelayBase            4
-
+#define PINRemoteImageManagerDefaultTimeout            30.0
+#define PINRemoteImageMaxRetries                       3
+#define PINRemoteImageRetryDelayBase                   4
+#define PINRemoteImageHTTPMaximumConnectionsPerHost    20
```

Comment:
Finally， I changed the HTTPMaximumConnectionsPerHost  value to equals UINT16_MAX.  

I read the answer from SO: http://stackoverflow.com/questions/2332741/what-is-the-theoretical-maximum-number-of-open-tcp-connections-that-a-modern-lin.

> Each TCP/IP packet has basically four fields for addressing; these are:  
> [source_ip][source_port][destination_ip][destination_port]
> Inside the TCP stack, these four fields are used as a compound key to match up packets to connections (e.g. file descriptors).  
> If a client has many connections to the same port on the same destination, then three of those fields will be the same - only source_port varies to differentiate the different connections. **Ports are 16-bit numbers**, therefore the maximum number of connections any given client can have to any given host port is 64K.

and this page: http://www.freesoft.org/CIE/Course/Section4/8.htm